src/python/platforms/android/adb.py

```diff
@@ -577,32 +538,81 @@ def reset_device_connection():
 
 def reset_usb():
   """Reset USB bus for a device serial."""
+
+  def _get_device_path():
```

Comment:
There is
environment.set_value('DEVICE_PATH', adb.get_device_path()) call, so can't move it inside.

Maybe a get_device_path() and that relies on get_device_path_using_usb and get_device_path_using_serial ?
then reset_usb just calls get_device_path ?

Reply:
Do we need that other call? It looks like the variable is only used here. It seems like it would be easy enough to factor this out so that it works safely, but it would be nice to remove this if we can.

Either way I think this function could use a bit more cleanup which I'll do once the question above is resolved.

Reply:
Discussed off-review. Needed to keep the other call but cleaned this up a bit more.